### PR TITLE
Allow printing of a function call, e.g. `print(abs(x))`

### LINKED ISSUE #1378: Can't print ufunc result
### Describe the bug
It is not possible to print the result of a NumPy `ufunc` directly and the error raised includes a traceback instead of a user friendly message.

### To Reproduce
Provide code to reproduce the behavior:
```python
def print_abs(x : 'int[:]'):
    print(abs(x))
```

### Error details
Provide the generated code, or the error message:
```bash
Traceback (most recent call last):
  File "/home/emily/.local/bin/pyccel", line 33, in <module>
    sys.exit(load_entry_point('pyccel', 'console_scripts', 'pyccel')())
  File "/home/emily/Code/pyccel/pyccel/commands/console.py", line 239, in pyccel
    execute_pyccel(filename,
  File "/home/emily/Code/pyccel/pyccel/codegen/pipeline.py", line 263, in execute_pyccel
    fname, prog_name = codegen.export(fname, language=language)
  File "/home/emily/Code/pyccel/pyccel/codegen/codegen.py", line 205, in export
    code = self._printer.doprint(self.ast)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/codeprinter.py", line 52, in doprint
    lines = self._print(expr).splitlines(True)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/codeprinter.py", line 88, in _print
    obj = getattr(self, print_method)(expr)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/ccode.py", line 691, in _print_Module
    body    = ''.join(self._print(i) for i in expr.body)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/ccode.py", line 691, in <genexpr>
    body    = ''.join(self._print(i) for i in expr.body)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/codeprinter.py", line 88, in _print
    obj = getattr(self, print_method)(expr)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/ccode.py", line 1601, in _print_FunctionDef
    body  = self._print(expr.body)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/codeprinter.py", line 88, in _print
    obj = getattr(self, print_method)(expr)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/ccode.py", line 1928, in _print_CodeBlock
    code = self._print(b)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/codeprinter.py", line 88, in _print
    obj = getattr(self, print_method)(expr)
  File "/home/emily/Code/pyccel/pyccel/codegen/printing/ccode.py", line 961, in _print_PythonPrint
    print_body = [ FunctionCallArgument(f[for_index]) ]
TypeError: 'NumpyAbs' object is not subscriptable
```

### Expected behavior
The printing stage should create an temporary variable allowing this object to be printed.

### Language
Fortran and C

### PR DESCRIPTION
**Pull request by sboof911**
Fix #1378:

- Create `tmp_var` variable to store the modified argument;
- Assign the value of the argument to `tmp_var` using the `assign_lhs_variable()` function;
- Append the assigned expression to `new_expression list`, then to `_additional_exprs list`;
- Replace the modified argument in `args list` with a `FunctionCallArgument` object using `tmp_var`.

### CODE REVIEW

**Comment by sboof911** [comment]
@EmilyBourne

**Comment by EmilyBourne** [comment]
Just to let you know, the PR description serves as a starting point to write the final commit message. It should be a concise summary of the changes you made, and why you made them so that other contributors can quickly understand your changes. You don't need to detail every step in your thought process otherwise some of the bigger PRs would require pages worth of description and no-one would read that :sweat_smile:

**Comment by EmilyBourne** [comment]
@sboof911 There seems to be a problem with GitHub's actions. While we are waiting for them to come back online, it is important that we keep the bot usage reasonable. Please only use the bot to run tests that you cannot run locally and to mark your PR as ready for review. It is extremely rare to need `/bot run pr_tests`. If you think it is ready, mark it. If you need to check if the tests run you can do this locally. Tests on other OSs only fail in special cases so it is not necessary to test them explicitly. You can wait until marking as "Ready for review" reveals a problem.

I have cancelled your test for now to allow higher priority tests to be run.

**Comment by EmilyBourne** [comment]
@sboof911 The bot is checking if you have finished your checklist here : https://github.com/pyccel/pyccel/pull/1404#issuecomment-1559088124

It will not let you mark as ready until everything is ticked

**Review by mkaddani — APPROVED** [review]
Looks good to me!

File: pyccel/parser/semantic.py [hunk 1]
```diff
@@ -3496,6 +3496,17 @@ def _visit_FunctionDef(self, expr):
 
     def _visit_PythonPrint(self, expr):
         args = [self._visit(i) for i in expr.expr]
+        for i, arg in enumerate(args):
+            if hasattr(arg.value, 'rank') and arg.value.rank and not isinstance(arg.value, (Variable, PythonTuple, IndexedElement)):
```
Comment (EmilyBourne) [inline pyccel/parser/semantic.py:3500]: Why not use the default argument for `getattr` to make this simpler:
```suggestion
            if getattr(arg.value, 'rank', 0) and not isinstance(arg.value, (Variable, PythonTuple, IndexedElement)):
```

**Review by EmilyBourne — CHANGES_REQUESTED** [review]
This looks good. I just have a few comments

**Review by EmilyBourne — CHANGES_REQUESTED** [review]
This looks much cleaner. But one of my previous comments has not been fixed yet

File: pyccel/parser/semantic.py [hunk 2]
```diff
@@ -3496,6 +3496,17 @@ def _visit_FunctionDef(self, expr):
...
+                new_symbol = PyccelSymbol(self.scope.get_new_name())
+                d_var = self._infer_type(arg.value)
+                new_expression = []
+                tmp_var = self._assign_lhs_variable(new_symbol, d_var , arg.value, new_expression , is_augassign=False)
+                creation = Assign(tmp_var, arg.value, fst=arg.value.fst)
+                new_expression.append(creation)
+                for v in new_expression:
+                    self._additional_exprs[-1].append(v)
+                args[i] = FunctionCallArgument(tmp_var)
```
[the remark this answers is not in the document]
Comment (sboof911) [reply pyccel/parser/semantic.py]: Fixed: d0b34daa5567bf77eb1b2fdeee9bc9648e26a17e

**Review by EmilyBourne — APPROVED** [review]

**Pull request merged**